### ISSUE TITLE
Interface not parsed

### ISSUE THREAD

**Issue by AlessandroMinoccheri** [phparkitect/arkitect#334]
### Bug Report

|    Q            |   A
|---------------- | ------
| BC Break        | no
| Library Version | 0.3.10
| PHP version     | 8.2

#### Summary

At the moment we are not parsing interfaces

#### How to reproduce

```
namespace App\Domain;
use Doctrine\ORM\QueryBuilder;
interface BookRepositoryInterface
{
    public function getBookList(): QueryBuilder;
}
```

With this config:

```
return static function (Config $config): void {
    $mvcClassSet = ClassSet::fromDir(__DIR__.'/src');

    $rules = [];

    $rules[] = Rule::allClasses()
        ->that(new ResideInOneOfTheseNamespaces('App\Domain'))
        ->should(new NotHaveDependencyOutsideNamespace('App\Domain'))
        ->because('we want protect our domain');

    $config
        ->add($mvcClassSet, ...$rules);
};
```

The tool passes all the tests but shouldn't!

#### Expected behavior
I am expecting that an error will throw for `BookRepositoryInterface` class

**Comment by LuigiCardamone** [comment]
There is a similar problem with the same config and this class:
```
class Foo
{
    public function foo(): ?QueryBuilder
    {
        return null;
    }
}
```

**Comment by AlessandroMinoccheri** [comment]
Closed by #335
